lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -233,13 +234,25 @@ class Chef::Knife::VsphereVmClone < Chef::Knife::BaseVsphereCommand
          description: 'Indicates whether to mark the new vm as a template',
          boolean: false
 
+  option :random_vmname,
+         long: '--random-vmname',
+         description: 'Creates a random VMNAME starts with vm-XXXXXXXX',
+         boolean: false
+
+  option :random_vmname_prefix,
+         long: '--random-vmname-prefix PREFIX',
+         description: 'Change the VMNAME prefix',
+         default: 'vm-'
+
   def run
     $stdout.sync = true
 
     vmname = @name_args[0]
-    if vmname.nil?
+    if (vmname.nil? && !config[:random_vmname]) || (vmname && config[:random_vmname])
```

Comment:
http://www.justinweiss.com/blog/2014/07/28/4-simple-memoization-patterns-in-ruby-and-one-gem/ is a good look at Ruby memoization/caching patterns.